lib/FrontendTool/FrontendTool.cpp

```diff
@@ -2209,13 +2218,15 @@ int swift::performFrontend(ArrayRef<const char *> Args,
           Instance->getMainModule()->getFiles());
     }
   }
+  // If we disabled the PrintingDiagnosticConsumer earlier in verify mode,
+  // reenable it now.
+  PDC.setSuppressOutput(false);
 
   if (diagOpts.VerifyMode != DiagnosticOptions::NoVerify) {
-    HadError = verifyDiagnostics(
-        Instance->getSourceMgr(),
-        Instance->getInputBufferIDs(),
-        diagOpts.VerifyMode == DiagnosticOptions::VerifyAndApplyFixes,
-        diagOpts.VerifyIgnoreUnknown);
+    // If we're in -verify mode, temporarily ignore errors. If verification
+    // failed, we'll still report an error correctly when finishDiagProcessing
+    // runs.
+    HadError = false;
```

Comment:
Is universally resetting the error code here a good idea? Granted, it's probably not a good idea for the verifier to have been doing this in the first place, but I imagine we've grown quite a few tests that rely on this behavior.

Reply:
I moved this logic into `finishDiagProcessing` instead so we override the error code at the last possible moment, which I think makes this logic easier to understand.